metricshub-engine/src/main/java/org/sentrysoftware/metricshub/engine/connector/model/common/EmbeddedFile.java

```diff
@@ -40,25 +45,25 @@ public class EmbeddedFile implements Serializable {
 	private static final long serialVersionUID = -197665338834839387L;
 
 	/**
-	 * The content of the embedded file.
+	 * The content of the embedded file as byte array.
 	 */
-	private String content;
+	private byte[] content;
 	/**
-	 * The type of the embedded file (e.g., MIME type).
+	 * The filename of the embedded file (e.g., script.bat).
 	 */
-	private String type;
+	private String filename;
 	/**
-	 * A reference to the embedded file, providing additional information or identification.
+	 * A reference to the embedded file, providing additional information for identification.
 	 */
-	private String reference;
+	private Integer id;
```

Comment:
What's the requirement for this `id`? Can we avoid it?

Reply:
This is the unique id of the embedded file, this will handle the case where different embedded files have the same name, and enhance slightly the memory of the global embedded file Map<Integer, EmbeddedFile>

Reply:
This is generated by the connector parser 😉.